lang/cs/Org.Apache.REEF.Client/YARN/IYarnREEFClient.cs

```diff
@@ -33,5 +34,14 @@ public interface IYarnREEFClient : IREEFClient
         /// <returns></returns>
         [Unstable("0.17", "Working in progress for rest API id returned")]
         Task<IReadOnlyDictionary<string, IApplicationReport>> GetApplicationReports();
+
+        /// <summary>
+        /// Kills the job application and returns Job status
+        /// If Application Id is invalid or application has been completed, exception will be thrown.
+        /// </summary>
+        /// <param name="appId">Application id to kill.</param>
+        /// <returns></returns>
+        [Unstable("0.17", "Working in progress for rest API status returned")]
+        Task<FinalState> KillJobApplication(string appId);
```

Comment:
As I said in the separate comment: This needs some error conditions. Specifically, define the `returns` in the xml doc and define what exceptions are thrown by this method.